plugins/yoga/src/create-yoga.ts

```diff
@@ -0,0 +1,76 @@
+import {
+  GraphQLOTELContext,
+  GraphQLOTELContextOptions,
+  TraceSchemaInput,
+  traceSchema,
+} from "@graphql-debugger/trace-schema";
+
+import { GraphQLSchema } from "graphql";
+import {
+  YogaServerInstance,
+  YogaServerOptions,
+  createYoga as originalCreateYoga,
+} from "graphql-yoga";
+
+interface DebuggerYogaOptions<TServerContext, TUserContext>
+  extends YogaServerOptions<TServerContext, TUserContext> {
+  debugger?: Omit<TraceSchemaInput, "schema"> & {
+    shouldDisable?: boolean;
+    otelContextOptions?: GraphQLOTELContextOptions;
+    traceSchemaOptions?: Omit<TraceSchemaInput, "schema">;
+  };
+  schema: GraphQLSchema;
+  context?: (req: any) => Promise<TUserContext>;
+}
+
+export function createYoga<
+  TServerContext extends Record<string, any> = {},
+  TUserContext extends Record<string, any> & {
+    GraphQLOTELContext?: GraphQLOTELContext;
+  } = {},
+>(
+  options: DebuggerYogaOptions<TServerContext, TUserContext>,
+): YogaServerInstance<TServerContext, TUserContext> {
+  if (!options.schema) {
```

Comment:
GraphQL Yoga's `schema` option is not always needed. It might be set by a plugin later. For example, Apollo Federation Gateway plugin and GraphQL Modules plugin set them without `options.schema`. Even if you set one, it overrides it.

Reply:
Within the context of graphql-debugger however, wouldn't it be helpful to have a behavior where we enforce a schema - since the Yoga object is under our control?

We could deal with edge cases as our users start using the app - and keep this comment in mind for the future. Or, do you think it's easier to do this work upfront? Just trying to get an idea :)

Reply:
It is not to enforce the schema :) I mean there are different ways to pass the schema to GraphQL Yoga. 
An existing GraphQL Yoga user might not be able to use your "wrapper" if a plugin passes the schema internally. 
Also in general, our plugin system is built exactly for this to avoid wrappers etc. You can easily access the internal lifecycle of GraphQL Yoga.

Reply:
It is true that a schema is required when using debugger as `traceSchema` will visit each resolver and wrap it in telemetry.

Going to hold up on merging and releasing this before further evaluating graphql yoga's plugin system.